### ISSUE TITLE
API for labeling and scenarios implementation

### ISSUE THREAD

**Issue by PK85** [kyma-incubator/compass#46]
**AC**
- document Label and LabelsDefinitons
  - how they work, how they depend each other
  - what LD delete does (is that possible when label exist)
  - what label key deletion means (removes all occurrances from apps and runtimes)
- implement LabelDefinition qGL Type and provide CRD (Update out of scope). Deletion removes a label from all entities (Apps and Runtimes). 
  - read also support getting list of schemas for given keys
  - provide LabelDefinition if you need something more than string. String is a default, and type is not required for addAppLabel and addRuntimeLabel in that case but internally new LabelDefiniton will be created. Has fields:
    - key
    - schema for value
- default scenario already exists, plus every app and runtime (if not specified) has that label in. LabelDefiniton already exists, we support only GET operation.

- implement `labels` qGL Type to mutate and query scenarios
  - create a scenario by adding new app or runtime with scenario label
  - create a scenario by labels endpoint
  - above ways are synced
  - scenarios for application and runtimes has same scope (could be reused)
- validation during Label creation
- remove annotations (we dont see a need for them right now)
- filtering by labels implemented on runtimes and applications

Out of scope:
- system specific labels. We can introduce that feature with getting scopes (together with tennat) when calling API.

**Comment by PK85** [comment]
To be discussed.
- We need to support system specific labels, which we use internal and which are not change able by the user like the user defined labels.
- The system specific label should have a registered type and probably json schema.  Overall new system label types can be added.
- The labels we show to the user and which he can use for his purpose are probably just one specific label type where name and value can only be a string.

See also: https://github.com/kyma-incubator/compass/issues/71

**Comment by PK85** [comment]
Expose LabelDefinition APIs asap to unblock UI tasks
